### ISSUE TITLE
Panics on searching for the empty string.

### ISSUE THREAD

**Issue by trishume** [jneem/memmem#1]
Currently in my code I have to have a separate case for if my search string is empty because `memmem` panics when using `search_in` with a searcher constructed with the empty string:

```
thread '<unnamed>' panicked at 'index 1 out of range for slice of length 0', src/libcore/slice.rs:513
```

It would be nice if when the searcher is an empty string search_in returns true iff the haystack is an empty string.

@jneem

**Comment by jneem** [comment]
Definitely a bug, thanks! But I think that if the needle is empty then it should always just match the beginning of the string, to make it consistent with the `std::str::find` method.

**Comment by jneem** [comment]
Fixed in 0.1.1.
